### PR TITLE
Fix delegation logic to check for expelled members

### PR DESCRIPTION
**Pull request by TomTaehoonKim**

### CODE REVIEW

**Review by Ham3798 — COMMENTED** [review]
LGTM  👍🏼👍🏼

File: core/src/reserved.rs [hunk 1]
```diff
@@ -913,6 +913,37 @@ mod tests {
...
+        state.members[0] = create_expelled_member(keys.clone(), 0);
+
+        // delegator: member-0000, delegatee: member-0002
+        let delegator = state.members[0].clone();
+        let delegator_private_key = keys[0].1.clone();
+        let delegatee = state.members[2].clone();
+
+        let data = DelegationTransactionData {
+            // delegator and delegatee are the same
+            delegator: delegator.name.clone(),
```
Comment (sigridjineth) [inline core/src/reserved.rs:929]: please fix this clippy to pass it

File: core/src/reserved.rs [hunk 2]
```diff
@@ -162,13 +176,13 @@ impl ReservedState {
         None
     }
 
-    pub fn is_member(&self, name: &MemberName) -> bool {
+    pub fn is_expelled(&self, name: &MemberName) -> Option<bool> {
```
Comment (sigridjineth) [inline core/src/reserved.rs:179]: why the `bool` is wrapped with `Option`?

**Review by sigridjineth — APPROVED** [review]

**Review by sejongk — APPROVED** [review]
LGTM!

File: core/src/reserved.rs (hunk 1 above)
Comment (TomTaehoonKim) [reply core/src/reserved.rs]: Applied. Thx.

**Review by junha1 — APPROVED** [review]

**Pull request merged**